pytests/tests/test_granular_perms.py

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright (C) 2022 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+
+import os
+import platform
+import pytest
+from subprocess import Popen, PIPE
+
+pkgname = None
+test_usr = 'test_usr'
+
+
+def run_bash_cmd(cmd, rc):
+    print(' '.join(cmd))
+    proc = Popen(cmd, stdout=PIPE, stderr=PIPE, stdin=PIPE)
+    out, err = proc.communicate()
+    out = out.decode().strip().split('\n')
+    err = err.decode().strip().split('\n')
+    print('stdout:', out, '\nstderr:', err)
+    assert('You have to be root' not in out)
+    assert('You have to be root' not in err)
+
+    if rc != 'ignore' and rc != (not not proc.returncode):
+        assert(False)
+
```

Comment:
Is there a test to check for sufficient permission when it's needed?

Reply:
Missed it. Will add this scenario as well.